extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalogAdapter.java

```diff
@@ -352,6 +353,163 @@ public Table listSnapshotsAsTable(@NotNull final String tableIdentifier) {
     }
 
+    /**
+     * Get a specific {@link Snapshot snapshot} of a given Iceberg table (or null if it does not exist).
+     *
+     * @param tableIdentifier The identifier of the table from which to gather snapshots
+     * @param snapshotId The id of the snapshot to retrieve
+     * @return The snapshot with the given id, or null if it does not exist
+     */
+    private Snapshot getSnapshot(@NotNull final TableIdentifier tableIdentifier, final long snapshotId) {
+        return listSnapshots(tableIdentifier).stream()
+                .filter(snapshot -> snapshot.snapshotId() == snapshotId)
+                .findFirst()
+                .orElse(null);
+    }
+
+    /**
+     * Get a legalized column rename map from a table schema and user instructions.
+     */
+    private Map<String, String> getRenameColumnMap(
+            @NotNull final org.apache.iceberg.Table table,
+            @NotNull final Schema schema,
+            @NotNull final IcebergInstructions instructions) {
+
+        final Set<String> takenNames = new HashSet<>();
+
+        // Map all the column names in the schema to their legalized names.
+        final Map<String, String> legalizedColumnRenames = new HashMap<>();
+
+        // Validate user-supplied names meet legalization instructions
+        for (final Map.Entry<String, String> entry : instructions.columnRenames().entrySet()) {
+            final String destinationName = entry.getValue();
+            if (!NameValidator.isValidColumnName(destinationName)) {
+                throw new TableDataException(
+                        String.format("%s - invalid column name provided (%s)", table, destinationName));
+            }
+            // Add these renames to the legalized list.
+            legalizedColumnRenames.put(entry.getKey(), destinationName);
+            takenNames.add(destinationName);
+        }
+
+        for (final Types.NestedField field : schema.columns()) {
+            final String name = field.name();
+            // Do we already have a valid rename for this column from the user or a partitioned column?
+            if (!legalizedColumnRenames.containsKey(name)) {
+                final String legalizedName =
+                        NameValidator.legalizeColumnName(name, s -> s.replace(" ", "_"), takenNames);
+                if (!legalizedName.equals(name)) {
+                    legalizedColumnRenames.put(name, legalizedName);
+                    takenNames.add(legalizedName);
+                }
+            }
+        }
+
+        return legalizedColumnRenames;
+    }
+
+    /**
+     * Return {@link TableDefinition table definition} for a given Iceberg table, with optional instructions for
+     * customizations while reading.
+     *
+     * @param tableIdentifier The identifier of the table to load
+     * @param instructions The instructions for customizations while reading
+     * @return The table definition
+     */
+    public TableDefinition getTableDefinition(
```

Comment:
Only the (String tableIdentifier, final long snapshotId) are implemented. The overloads for TableIdentifier, Snapshot are not implemented.

Reply:
We want the remainder, right? @devinrsmith @lbooker42

Reply:
Yes, will add them. needing to support every variant of these arguments does dispose me toward an overhaul of the API.